app/controllers/items_controller.rb

```diff
@@ -1,6 +1,6 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!, except: [:index, :show]
-  before_action :set_item, only: [:show, :edit, :update]
+  before_action :set_item, only: [:show, :edit, :update, :destroy]
   before_action :move_to_index, only: [:edit, :update]
```

Comment:
```suggestion
  before_action :move_to_index, only: [:edit, :update]
```
出品者でなければ削除できないようにしましょう。
詳細ページにおける「削除」ボタンの表示・非表示に加え、コントローラー側でも制御します。